File: VB/DXExample.Module/HidePivotGridFieldsControllerBase.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.PivotChart

Namespace DXExample.Module
	Public MustInherit Class HidePivotGridFieldsControllerBase
		Inherits ViewController
		Public Sub New()
			TargetObjectType = GetType(IAnalysisInfo)
			TargetViewType = ViewType.DetailView
		End Sub
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			AddHandler View.ControlsCreated, AddressOf View_ControlsCreated
		End Sub
		Private Sub View_ControlsCreated(ByVal sender As Object, ByVal e As EventArgs)
			For Each editor As AnalysisEditorBase In (CType(View, DetailView)).GetItems(Of AnalysisEditorBase)()
				SetFieldBuider(editor.Control)
			Next editor
		End Sub
		Protected MustOverride Sub SetFieldBuider(ByVal control As IAnalysisControl)
	End Class
End Namespace

File: CS/DXExample.Module/HidePivotGridFieldsControllerBase.cs
using System;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.PivotChart;

namespace DXExample.Module {
    public abstract class HidePivotGridFieldsControllerBase : ViewController {
        public HidePivotGridFieldsControllerBase() {
            TargetObjectType = typeof(IAnalysisInfo);
            TargetViewType = ViewType.DetailView;
        }
        protected override void OnActivated() {
            base.OnActivated();
            View.ControlsCreated += new EventHandler(View_ControlsCreated);
        }
        private void View_ControlsCreated(object sender, EventArgs e) {
            foreach(AnalysisEditorBase editor in ((DetailView)View).GetItems<AnalysisEditorBase>()){
                SetFieldBuider(editor.Control);
            }
        }
        protected abstract void SetFieldBuider(IAnalysisControl control);
    }
}
